external/emulation/emulation/masks.py

```diff
@@ -37,3 +37,17 @@ def __call__(self, state: FortranState, emulator: FortranState) -> FortranState:
             out[self.key] = np.minimum(out[self.key], self.max)
 
         return out
+
+
+class LevelMask:
+    def __init__(self, key: str, start: Optional[int], stop: Optional[int]):
+        self.key = key
+        self.start = start
+        self.stop = stop
+
+    def __call__(self, state: FortranState, emulator: FortranState) -> FortranState:
+        out = {**emulator}
+        use_fortran_state = slice(self.start, self.stop)
+        # Fortran state TOA is index 79, and dims are [z, sample]
+        out[self.key][use_fortran_state] = state[self.key][use_fortran_state]
```

Comment:
This will actually mutate `state[key]` since it points to the same numpy array as `out[self.key]`. Either do a copy or use `np.where`.

Reply:
Updated to remove the input argument mutation.